multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/BindServiceToApplicationStep.java

```diff
@@ -18,29 +20,47 @@
 import com.sap.cloudfoundry.client.facade.ApplicationServicesUpdateCallback;
 import com.sap.cloudfoundry.client.facade.CloudControllerClient;
 import com.sap.cloudfoundry.client.facade.CloudOperationException;
+import com.sap.cloudfoundry.client.facade.domain.CloudServiceBinding;
+import com.sap.cloudfoundry.client.facade.domain.ServiceCredentialBindingOperation;
 
 @Named("bindServiceToApplicationStep")
 @Scope(BeanDefinition.SCOPE_PROTOTYPE)
-public class BindServiceToApplicationStep extends SyncFlowableStep {
+public class BindServiceToApplicationStep extends AsyncFlowableStep {
 
     @Override
-    protected StepPhase executeStep(ProcessContext context) {
+    protected StepPhase executeAsyncStep(ProcessContext context) throws Exception {
         CloudApplicationExtended app = context.getVariable(Variables.APP_TO_PROCESS);
         String service = context.getVariable(Variables.SERVICE_TO_UNBIND_BIND);
         Map<String, Object> serviceBindingParameters = context.getVariable(Variables.SERVICE_BINDING_PARAMETERS);
-
         getStepLogger().info(Messages.BINDING_SERVICE_INSTANCE_0_TO_APPLICATION_1, service, app.getName());
+        CloudControllerClient controllerClient = context.getControllerClient();
+        Optional<String> jobId = controllerClient.bindServiceInstance(app.getName(), service, serviceBindingParameters,
+                                                                      getApplicationServicesUpdateCallback(context, controllerClient));
+        if (context.getVariable(Variables.USE_LAST_OPERATION_FOR_SERVICE_BINDING_CREATION)) {
```

Comment:
This will be used when the controller returns 422 and the binding exists. Then will exit the step, and when creating the polling executions, the factory that creates them will decide which polling execution to create - "poll by async job" or "poll by last operation"